Source/Core/Core/HW/DVDInterface.cpp

```diff
@@ -1244,30 +1253,28 @@ void ScheduleReads(u64 dvd_offset, u32 dvd_length, bool decrypt, u32 output_addr
             buffer_start, buffer_end, buffer_end - buffer_start);
 
   DEBUG_LOG(DVDINTERFACE,
-            "Schedule reads: offset=0x%" PRIx64 " length=0x%" PRIx32 " address=0x%" PRIx32,
-            dvd_offset, dvd_length, output_address);
+            "Schedule reads: offset=0x%" PRIx64 " length=0x%" PRIx32 " address=0x%" PRIx32, offset,
+            length, output_address);
 
   // The DVD drive's minimum turnaround time on a command, based on a hardware test.
   s64 ticks_until_completion = COMMAND_LATENCY_US * (SystemTimers::GetTicksPerSecond() / 1000000);
 
   u32 buffered_blocks = 0;
   u32 unbuffered_blocks = 0;
 
-  while (dvd_length > 0)
+  while (length > 0)
   {
-    // Where the read actually takes place on disc
-    u64 rounded_offset = Common::AlignDown(dvd_offset, DVD_ECC_BLOCK_SIZE);
+    const u32 bytes_per_chunk =
+        decrypt ? DiscIO::CVolumeWiiCrypted::BLOCK_DATA_SIZE : DVD_ECC_BLOCK_SIZE;
```

Comment:
In a Wii partition, each 32 KiB block contains 31 KiB of data that actually gets passed to the game and 1 KiB of data that only IOS uses. This means that 1 GiB of data on the DVD only contains about 0.97 GiB of data that the game uses. Since the `length` parameter that's passed in to `ScheduleReads` uses the addressing scheme where that would be 0.97 GiB, we have to convert it to actual DVD offsets, as we otherwise will get wrong values for both `length` and `speed` in `CalculateRawDiscReadTime`.

I'm currently not modeling any decryption delay.